AyonCiCd/Project.py

```diff
@@ -435,24 +435,23 @@ def _install_pip_packages_in_venv(self, venv_path: str, pip_package_list: list):
             pip_install_list = " ".join(pip_package_list)
             pip_install_command = f"pip install {pip_install_list}"
 
-        command = []
+        command = f"{self.__get_venv_activate_cmd(venv_path)} && pip install --upgrade pip"
+        if pip_install_command:
+            command += f" && {pip_install_command}"
+
         if sys.platform.lower() == "win32":
-            command = f'cmd /c "{self.__get_venv_activate_cmd(venv_path)} && pip install --upgrade pip"'
-            if pip_install_command:
-                command = f'cmd /c "{self.__get_venv_activate_cmd(venv_path)} && pip install --upgrade pip && {pip_install_command}"'
-        else:
-            command = [f"{self.__get_venv_activate_cmd(venv_path)} && pip install --upgrade pip"]
-            if pip_install_command:
-                command = [f"{self.__get_venv_activate_cmd(venv_path)} && pip install --upgrade pip && {pip_install_command}"]
+            command = f'cmd /c "{command}"'
 
-        process = subprocess.Popen(
-            command,
-            shell=True,
-        )
+        try:
+            process = subprocess.check_call(
+                command,
+                shell=True,
+            )
+
+        except subprocess.CalledProcessError as error:
+            # TODO log failed command and log warning
```

Comment:
I'd recommand `check_call` instead of `Popen` there which does the returncode check for you.
I think it's important to log properly the failing command and error result (in terminal and in output log file). 

Initially I had quite a lot of issues related to CMAKE (but nothing to do with AYON) and I found it not too easy to see what was going on.

Reply:
agreed sounds good lets do it this way. 
i will test this version on the cpp api.